reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerOperations.java

```diff
@@ -814,6 +814,10 @@ else if (msg instanceof HttpRequest) {
 				}
 				else {
 					request.release();
+					// HTTP/1.1 TLS Upgrade (RFC-2817) on empty requests (GET/HEAD/OPTIONS)
+					if (!isHttp2() && request.headers().contains(HttpHeaderNames.UPGRADE)) {
+						onInboundComplete();
```

Comment:
I think we don't need to call `channel().config().setAutoRead(true);` for HTTP/1.1

Reply:
We always switch to auto-read=true between the different requests so I would prefer to have it here.